src/serialiser.c

```diff
@@ -214,6 +214,9 @@ mqtt_serialiser_rc_t mqtt_serialiser_write(mqtt_serialiser_t* serialiser, mqtt_m
   do {
     buffer[offset++] = remaining_length & 0x7f;
     remaining_length >>= 7;
+    // Set the high bit to indicate more bits
+    if(remaining_length > 0)
```

Comment:
@andreas-ibm Please, put opening and closing curly braces around the if condition.